src/backend/optimizer/plan/planner.c

```diff
@@ -1233,11 +1236,31 @@ inheritance_planner(PlannerInfo *root)
 	/*
 	 * And now we can get on with generating a plan for each child table.
 	 */
+
+	/*
+	 * Make a working copy of the PlannerInfo and fill RelOptInfo
+	 * arrays to check later if the child need to be scanned.
+	 */
+	fakeroot = palloc(sizeof(PlannerInfo));
+	memcpy(fakeroot, root, sizeof(PlannerInfo));
```

Comment:
I suggest replacing memcpy with "*fakeroot = *root;". The compiler will check types of variables and the type mismatch error will be detected automatically. The error of incorrectly specifying the size of the structure will not be possible. The compiler will be able to generate more optimal code